### ISSUE TITLE
High Contrast: Firefox 78.13.0 ESR: NumberInput border is cut off at the inc/dec control when in high contrast mode

### ISSUE THREAD

**Issue by empoon** [carbon-design-system/carbon#9548]
## Environment
Tested with Carbon React sample page with latest carbon version and Carbon Story booking using version 7.18.0

> Operating system
windows 10, high contrast mode
> Browser
Firefox 78.13.0ESR

## Detailed description

> What version of the Carbon Design System are you using?
 "carbon-components": "10.17.0",
 "carbon-components-react": "7.17.0",
 "carbon-icons": "^7.0.7",
> What did you expect to happen?
the border line should be around the whole number Input area
> What happened instead?
the border line is cut off at the increment and decrement control. See screenshot
![image](https://user-images.githubusercontent.com/89608071/131026477-c617f3d3-d1cf-423e-85de-927734c56df5.png)
same problem can be observer with newer version of carbon
![image](https://user-images.githubusercontent.com/89608071/131026931-7ec4ddc2-1ea7-4def-ac77-f1041b15ec86.png)

## Steps to reproduce the issue

1. On windows, turn on High Contrast and select any mode
2. Launch latest Firefox ESR version, open the carbon sample page for numberInput
result: Look at the border of the numberInput and it gets cut off

**Comment by aledavila** [comment]
@empoon can you confirm the above please ^

**Comment by empoon** [comment]
Hi, yes the issue occurs with the latest numberInput in storybook as well.
